### PR TITLE
feat: add reset settings button

### PR DESCRIPTION
**Pull request by oliverqx**
/claim #838

## description
adds reset settings button

related issue: #838

## how to test

add a few steps to test the pr in the most time efficient way.

1. go to settings
2. scroll to bottom
3. click red button

design inspired in vercel's sensitive task button

Screenshot 2024-12-04 at 14 04 48

### CODE REVIEW

**Comment by oliverqx** [comment]
looks like theres an issue with macos path permissons, im working on it

**Comment by oliverqx** [comment]
@louis030195 i think its working now, cant test if capabilities were properly set for windows

**Comment by oliverqx** [comment]
definitely working now, deleting the store is not necessary and i couldnt know for sure if it works on windows

File: screenpipe-app-tauri/lib/hooks/use-settings.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { platform } from "@tauri-apps/plugin-os";
 import { Pipe } from "./use-pipes";
 import posthog from "posthog-js";
 import { Language } from "@/lib/language";
+import { remove } from "@tauri-apps/plugin-fs";
```
Comment (tribhuwan-kumar) [inline screenpipe-app-tauri/lib/hooks/use-settings.tsx:8]: @oliverqx unused import??

File: screenpipe-app-tauri/lib/hooks/use-settings.tsx [hunk 2]
```diff
@@ -140,213 +140,218 @@ export function useSettings() {
     }
   };
 
-  useEffect(() => {
-    const loadSettings = async () => {
-      if (!store) {
-        await initStore();
-      }
+  const resetSettings = async () => {
```
Comment (tribhuwan-kumar) [inline screenpipe-app-tauri/lib/hooks/use-settings.tsx:143]: @oliverqx you're just changing settings to default values of settings but some values are meant to be set dynamically as an example the `fps` value in `defaultSettings` object is 0.5 and this `resetSettings` function is going to set the `fps` value `0.5` on all os but it should change the values of fps dynamically depending upon os

```typescript
        const savedFps =
          (await store!.get<number>("fps")) ||
          (currentPlatform === "macos" ? 0.2 : 1);
```

also there are some unused imports!

**Comment by oliverqx** [comment]
I also wanted to clean up the loadSettings function like this:

```
async function createUserSettings(
  store: Store,
  currentPlatform: string
): Promise<Settings> {
  let defaultSettingsObject = createDefaultSettingsObject(currentPlatform)

  let userSettingsObject: Record<string,any> = {}

  for (const key of Object.keys(defaultSettingsObject)) {
    userSettingsObject[key as keyof Settings] = 
      await store.get(key) || defaultSettingsObject[key as keyof Settings]
  }

  return userSettingsObject as Settings
}
```


```
  const loadSettings = async () => {
    if (!store) {
      await initStore();
    }

    try {
      const currentPlatform = platform();
      const userSettings = await createUserSettings(
        store!,
        currentPlatform
      )

      setSettings(userSettings);
    } catch (error) {
      console.error("failed to load settings:", error);
      setSettings((prevSettings) => ({ ...prevSettings }));
    }
  };
```

**Comment by louis030195** [comment]
nice!

is it done?

**Comment by oliverqx** [comment]
yes its fully working now

just recorded a loom video:
https://www.loom.com/share/3cad2609599745ea82d7892f7e630bdd?sid=4f648418-4d93-48dc-889f-acb370d285f5

**Comment by oliverqx** [comment]
thanks!

/claim #838

**Comment by oliverqx** [comment]
I think i need to create another pr to claim, was thinking i can do some no-op change somwhere and open a pr.

**Comment by oliverqx** [comment]
@louis030195 hi there just wanted to let you know thatit looks like the bounty was not rewarded. i think there is a step missing

**Comment by louis030195** [comment]
@oliverqx fyi settings are broken since that merge and IDK how to fix - it just does not update anymore

**Comment by oliverqx** [comment]
@louis030195 

checking rn

**Pull request merged**